src/backend/distributed/planner/modify_planner.c

```diff
@@ -641,3 +644,331 @@ ExtractPartitionValue(Query *query, Var *partitionColumn)
 
 	return partitionValue;
 }
+
+
+/*
+ * ContainsPartitionColumnMatchExpression returns true if the query contains an exact
+ * match (equal) expression on the partition column. The function returns true only
+ * if the match expression has an AND relation with the rest of the expression tree.
+ */
+static bool
+ContainsPartitionColumnMatchExpression(Node *node, Var *partitionColumn)
+{
+	if (node == NULL)
+	{
+		return  false;
+	}
+
+	if (IsA(node, OpExpr))
+	{
+		OpExpr *opExpr = (OpExpr *) node;
+		bool simpleExpression = SimpleOpExpression((Expr *) opExpr);
+		bool partitionColumnInExpr = false;
+		char *operatorName  = NULL;
+		int equalsCompare = -1;
+		bool usingEqualityOperator = false;
+
+		if (!simpleExpression)
+		{
+			return false;
+		}
+
+		partitionColumnInExpr = OpExpressionContainsColumn(opExpr, partitionColumn);
+		if (!partitionColumnInExpr)
+		{
+			return false;
+		}
+
+		operatorName = get_opname(opExpr->opno);
```

Comment:
So, I realized that there is some other way of checking operator name instead of a text check:

```C

OpExpr *partColumnEqualityExpr = MakeOpExpression(partitionColumn, BTEqualStrategyNumber);
if (equality_ops_are_compatible(partColumnEqualityExpr->opno, opExpr->opno) )
{
... do what's needed here
}
```

I'm bringing this up, because I'm not comfortable with checking an operator by it's name. Do you think what I suggest work for us?